code/game/machinery/multistructure.dm

```diff
@@ -0,0 +1,147 @@
+/*
+	Multistructures consist of ingame objects in certain placement
+	when all object in right place and last one is constucted multistructure datum is created and links all elements
+
+	strucure is constructed -> check if multistructure can be created checkMS() -> if yes adds all structures in multistructure elements var ->
+	-> then multistructure init() and creates links between multistructure and elements
+
+	All interactions between machines should be done in multistructure datum
+
+
+
+*/
+
+
+//	This proc will try to find multistrucure starting in given coords
+//	You pass coords of where should be top-left element of MS structure matrix
+//	It will check if all those elements in right order and return true
+/proc/is_multistructure(var/x, var/y, var/z, var/list/structure)
+	if(!x || !y || !z || !structure)
+		error("Passed wrong arguments to is_multistructure()")
+		return FALSE
+	for(var/i = 1 to structure.len)
+		var/list/row = structure[i]
+		for(var/j = 1 to row.len)
+			if(!(locate(row[j]) in locate(x + (j-1),y - (i-1),z)))
+				return FALSE
+	return TRUE
+
+
+//	Creates multistructure datum and connects it to all elements mentioned in multistructure
+//	You pass coords of where should be top-left element of MS structure matrix
+//	It will check if all those elements in right order using is_multistructure() and then proceed to create multistructure
+/proc/create_multistructure(var/x, var/y, var/z, var/datum/multistructure/MS)
+	if(!x || !y || !z || !istype(MS))
+		error("Passed wrong arguments to create_multistructure()")
+		return FALSE
+	if(!is_multistructure(x, y, z, MS.structure))
+		return FALSE
+	for(var/i = 1 to MS.structure.len)
+		var/list/row = MS.structure[i]
+		for(var/j = 1 to row.len)
+			var/obj/machinery/multistructure/M = locate(row[j]) in locate(x + (j-1),y - (i-1),z)
+			MS.elements += M
+	MS.init()
+	if(MS.validate())
+		return TRUE
+	return FALSE
+
+
+/datum/multistructure
+	var/list/structure = list()
+	var/list/elements = list()
+
+
+/datum/multistructure/Destroy()
+	disconnect_elements()
+	return ..()
+
+
+/datum/multistructure/proc/init()
```

Comment:
is this typo or not?
didn't we use `initialize()` instead of standart `init()` or it's created only for objects?

Reply:
Yep, for internal purposes